projects/btldr/py_client/btldr_database.py

```diff
@@ -0,0 +1,250 @@
+from cantools.database.can import (
+    Message as CANMessage,
+    Signal as MessageSignal,
+    Database as DBC,
+)
+
+
+class BtldrDatabase(DBC):
+    def __init__(self, ecu_id=0):
+        super().__init__()
+
+        self.ecu_id = ecu_id
+
+        self._messages.append(self._make_query_msg())
+        self._messages.append(self._make_query_response_msg())
+        self._messages.append(self._make_reset_msg())
+        self._messages.append(self._make_reset_response_msg())
+        self._messages.append(self._make_request_msg())
+        self._messages.append(self._make_request_response_msg())
+        self._messages.append(self._make_data_msg())
+        self._messages.append(self._make_data_response_msg())
+
+        self.refresh()
+
+    def _make_query_msg(self) -> CANMessage:
+        timestamp = MessageSignal(
+            "timestamp", start=0, length=64, unit="seconds", scale=1, offset=0
+        )
+
+        query = CANMessage(
+            frame_id=self.ecu_id + 0,
+            name="btldr_query",
+            length=8,
```

Comment:
nit: abstract out the message lengths and start bits into variables to make them easier to change if we ever want to change the messages